src/vs/workbench/contrib/accessibilitySignals/browser/accessibilitySignalLineFeatureContribution.ts

```diff
@@ -215,20 +233,22 @@ class FoldedAreaLineFeature implements LineFeature {
 	createSource(editor: ICodeEditor, _model: ITextModel): LineFeatureSource {
 		const foldingController = FoldingController.get(editor);
 		if (!foldingController) {
-			return { isPresent: () => false, };
+			return { isPresent: () => false, isPresentOnLine: () => false };
 		}
-		const foldingModel = observableFromPromise(foldingController.getFoldingModel() ?? Promise.resolve(undefined));
-		return {
-			isPresent: (position, reader) => {
+		class FoldedAreaLineFeatureSource extends LineFeatureSource {
```

Comment:
In order to make use of the default `isPresent`, I had to implement these as classes

See https://github.com/microsoft/vscode/pull/209066/commits/ecf7466e6b7fe57542412f0688033dba3f17417f for alternate approach.

Reply:
maybe `isPresent` should just be optional instead?

Reply:
Aha, I see. Yes, I like your latest change more!